js/core.js

```diff
@@ -642,7 +642,9 @@ var ERMrest = (function (module) {
             var uri = this._getBaseURI(api);
 
             if (filter !== undefined && filter !== null) {
-                uri = uri + "/" + filter.toUri();
+                var filterUri = filter.toUri();
+                // Set filter in url only if the filterURI is not empty
+                if (filterUri.trim().length) uri = uri + "/" + filterUri
```

Comment:
Thats what I am trying to check, if the string length is 0, then don't send the filter.